core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -964,14 +981,27 @@ public TerminationHandle warmTopReader(final IndexShard indexShard, IndexMetaDat
             final Map<String, MappedFieldType> warmUpGlobalOrdinals = new HashMap<>();
             for (DocumentMapper docMapper : mapperService.docMappers(false)) {
                 for (FieldMapper fieldMapper : docMapper.mappers()) {
-                    final FieldDataType fieldDataType = fieldMapper.fieldType().fieldDataType();
+                    final FieldDataType fieldDataType;
+                    final String indexName;
+                    if (fieldMapper instanceof ParentFieldMapper) {
+                        MappedFieldType joinFieldType = ((ParentFieldMapper) fieldMapper).joinFieldType();
+                        if (joinFieldType == null) {
+                            continue;
+                        }
+                        fieldDataType = joinFieldType.fieldDataType();
```

Comment:
no, it is added via the child type's childJoinFieldType. If there is an active child parent field it is added via that parent field instance (childType.childJoinFieldType == parentType. parentJoinFieldType)